synapse/util/retryutils.py

```diff
@@ -172,6 +177,7 @@ def __exit__(self, exc_type, exc_val, exc_tb: TracebackType) -> None:
             # failures; this is mostly so as not to catch defer._DefGen.
             valid_err_code = True
         elif issubclass(exc_type, CodeMessageException):
+            exc_val_cme = cast(CodeMessageException, exc_val)
```

Comment:
We may as well replace the `issubclass` with an `isinstance(exc_val, CodeMessageException)` rather than casting?